api/api.go

```diff
@@ -210,10 +210,15 @@ func New(
 	verboseLogging uint, authType uint8,
 	opts *ClientOptions) (Client, error) {
 
-	if hostname == "" || username == "" || password == "" || authType > 1 {
+	if hostname == "" || username == "" || password == "" {
 		return nil, errNewClient
 	}
 
+	if authType != 0 && authType != 1 {
```

Comment:
compare against consts authTypeBasic and authTypeSessionBased

Reply:
Updated